### ISSUE TITLE
Square rework required

### ISSUE THREAD

**Issue by scottcwilson** [zencart/zencart#4679]
Per Square: We wanted to remind you that the SqPaymentForm JavaScript library will be retired on 07/21/2022 and its functionality will be shut off. SqPaymentForm has been replaced by the Web Payments SDK and we encourage you to migrate as soon as possible by following this step-by-step migration guide.

**Comment by brittainmark** [comment]
Can we make sure SCA compliance is included.

**Comment by scottcwilson** [comment]
> Can we make sure SCA compliance is included.

Notes on doing this from the current implementation: 
https://github.com/zencart/zencart/pull/4629

**Comment by brittainmark** [comment]
Square website [https://developer.squareup.com/docs/sca-overview](https://developer.squareup.com/docs/sca-overview) and [https://developer.squareup.com/docs/web-payments/sca](https://developer.squareup.com/docs/web-payments/sca)

Basically, you need to add a verify buyer step. This is fine for all countries, even if SCA is not required. In non SCA countries, no authorisation is requested from the buyer.

You can see the changes I made to make it work in this zip file [SquareSca](https://inner-light.co.uk/SquareSCA.zip) This correctly shows the invalid credit card entries unlike @rocketsites version which was in #4629 it can be downloaded from [RocketSquare](https://inner-light.co.uk/square.zip)

If you want, I can branch 1.5.8 and add my version to my git hub for easy comparison.

Things to note. Neither version copes with the person not ticking the accept condition box. In this scenario, all the credit card details are lost. At the moment, if SCA fails, you are returned to the payment details page (re-enter credit card details) all credit card details are blank again on the form.

**Comment by proseLA** [comment]
https://www.dropbox.com/s/b2pabxh4tk6q1rh/Screencast%20from%2003-16-2022%2007%3A36%3A52%20AM.webm?dl=0

**Comment by scottcwilson** [comment]
Fantastic progress!

**Comment by brittainmark** [comment]
Do you want any testing done?

**Comment by scottcwilson** [comment]
Please wait until the work is done.

**Comment by proseLA** [comment]
> Excellent is that with the new web payment method as well.

i believe its called the web payments SDK.  and yes, it does use that SDK.  

> Do you want any testing done?

still a ways away from needing outside testing...  but will keep it in mind.

**Comment by scottcwilson** [comment]
Closing this ticket now that we have a plugin at 
https://www.zen-cart.com/downloads.php?do=file&id=2345

**Comment by lat9** [comment]
Given that the plugin works and the `square` payment method included in the Zen Cart base doesn't, shouldn't the base version be removed from the distribution?  We could add informative text to the zc158 distribution's documentation to identify what needs to happen on an upgrade.

**Comment by scottcwilson** [comment]
Yes, the base version will be removed.  And yes the doc needs work - agreed.

**Comment by lat9** [comment]
Sounds good; closing.
